x/stakeibc/keeper/msg_server_create_trade_route_test.go

```diff
@@ -0,0 +1,231 @@
+package keeper_test
+
+import (
+	sdkmath "cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	icatypes "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/types"
+
+	"github.com/Stride-Labs/stride/v16/x/stakeibc/keeper"
+	"github.com/Stride-Labs/stride/v16/x/stakeibc/types"
+)
+
+func (s *KeeperTestSuite) SetupTestCreateTradeRoute() (msg types.MsgCreateTradeRoute, expectedTradeRoute types.TradeRoute) {
+	rewardChainId := "reward-0"
+	tradeChainId := "trade-0"
+
+	hostConnectionId := "connection-0"
+	rewardConnectionId := "connection-1"
+	tradeConnectionId := "connection-2"
+
+	hostToRewardChannelId := "channel-100"
+	rewardToTradeChannelId := "channel-200"
+	tradeToHostChannelId := "channel-300"
+
+	rewardDenomOnHost := "ibc/reward-on-host"
+	rewardDenomOnReward := RewardDenom
+	rewardDenomOnTrade := "ibc/reward-on-trade"
+	hostDenomOnTrade := "ibc/host-on-trade"
+	hostDenomOnHost := HostDenom
+
+	withdrawalAddress := "withdrawal-address"
+	unwindAddress := "unwind-address"
+
+	poolId := uint64(100)
+	maxAllowedSwapLossRate := "0.05"
+	minSwapAmount := sdkmath.NewInt(100)
+	maxSwapAmount := sdkmath.NewInt(1_000)
+
+	// Mock out connections for the reward an trade chain so that an ICA registration can be submitted
+	s.MockClientAndConnection(rewardChainId, "07-tendermint-0", rewardConnectionId)
+	s.MockClientAndConnection(tradeChainId, "07-tendermint-1", tradeConnectionId)
+
+	// Register an exisiting ICA account for the unwind ICA to test that
+	// existing accounts are re-used
+	// TODO [DYDX]: Replace with trade route owner
+	owner := types.FormatICAAccountOwner(rewardChainId, types.ICAAccountType_CONVERTER_UNWIND)
+	s.MockICAChannel(rewardConnectionId, "channel-0", owner, unwindAddress)
+
+	// Create a host zone with an exisiting withdrawal address
```

Comment:
The address exists but we never called MockICAChannel on the withdrawal address so we can't send messages to it yet

Reply:
Ok I understand better now, this function is just trying to setup the new trade route proposal and the expected outcome, so we don't need any of these connections to actually be real beyond generating the owner and ports as expected inside the setup.  The withdrawal account will be setup from the hostZone while the unwind and trade will be setup with these mock helpers.

Reply:
exactly!